### PR TITLE
ADR: Development Environments

### PR DESCRIPTION
**Pull request by carltonsmith**
### Summary of Changes

Documents proposed change in our dev environment strategy for Cloud.gov.

### CODE REVIEW

**Review by kniz-raft — COMMENTED** [review]
@carltonsmith Would it be possible to include approximate effort for this lift? I believe you included that in another one and I find it helpful.

**Comment by jtwillis92** [comment]
Added a time estimate of 4-6 hours - the biggest obstacle to implementing this is the need to pass in something to Circle CI from the GitHub action that determines which instance to deploy to based on the associated label. To work around that I'd suggest using different `deploy` jobs in CircleCI for each option - which could then be called from the GitHub action by the string name and avoid the need to do some of the conditional logic we have been doing in the Circle CI config.

**Review by jtwillis92 — APPROVED** [review]

**Comment by ADPennington** [comment]
@carltonsmith one question we have on this one is about space: do we have a sense of how much is needed for each of the 4 new environments? currently there are 8 apps (4 of these seem like they are no longer in use, 2 are for staging, and 2 are for dev), so it doesn't appear like this would present an issue with the # of apps (assuming there's some cleanup here), but we'd like to discuss more if these additions will keep us within our existing memory quota. cc: @lfrohlich @alexsoble

**Comment by carltonsmith** [comment]
> @carltonsmith one question we have on this one is about space: do we have a sense of how much is needed for each of the 4 new environments? currently there are 8 apps (4 of these seem like they are no longer in use, 2 are for staging, and 2 are for dev), so it doesn't appear like this would present an issue with the # of apps (assuming there's some cleanup here), but we'd like to discuss more if these additions will keep us within our existing memory quota. cc: @lfrohlich @alexsoble

Yes, we won't have an issue with memory or space with this approach. We are also discussing adjusting the space and memory values for each app, as right now we have more allocated than is needed for both the frontend and the backend.

**Comment by carltonsmith** [comment]
@ADPennington @alexsoble @lfrohlich I have updated the ADR with disk space allocation information as we discussed in Tech Sync on Friday

**Comment by ADPennington** [comment]
> @ADPennington @alexsoble @lfrohlich I have updated the ADR with disk space allocation information as we discussed in Tech Sync on Friday

thanks for adding this @carltonsmith! quick question: do we have a sense of how much disk space would be allocated to each of these 4 dev environments?

**Comment by carltonsmith** [comment]
> thanks for adding this @carltonsmith! quick question: do we have a sense of how much disk space would be allocated to each of these 4 dev environments?

Yes, it's in the ADR in [this section](https://github.com/raft-tech/TANF-app/pull/825/files#diff-c253b7c5dcc627b79806aae524d52cb27356b16f4ddf43b419a84266aa1f918eR27) @ADPennington

File: docs/Architecture Decision Record/015-development-environments.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# 15. Development Environments
+
+Date: 2021-04-06 (yyyy-mm-dd)
+
+## Status
+
+Pending
```
Comment (ADPennington) [inline docs/Architecture Decision Record/015-development-environments.md:7]:
```suggestion
Approved
```

**Review by ADPennington — APPROVED** [review]
thanks @carltonsmith. after a little perusing around in cloud.gov with @alexsoble, we didn't find any evidence that there is a disk space limit, but there is limit for memory, so we'll want to keep an eye on that. It doesn't seem like an issue right now (e.g. 7 apps running now and under the memory quota). and tearing down apps not in use sounds like a promising option to consider to help keep us within the mem quota if it does become a problem. cc: @lfrohlich @alexsoble

**Pull request merged**